### ISSUE TITLE
Custom methods don't work with hooks

### ISSUE THREAD

**Issue by leob** [arve0/feathers-custom-methods#4]
Very basic use case: if I define a custom method on a service and then I try to configure hooks on the service, Feathers throws an error that the method isn't supported.

What I saw in the Feathers source code is that the supported methods are sort of 'hardcoded'. This is done within a "plugin" which is executed during app setup, and the allowed methods are configured globally on the app level. It looks fairly inflexible, don't know why it was done this way ... it also looks like it's not easy to replace this plugin or override its behavior.

Did this work pre-Buzzard (Auk)?

**Comment by arve0** [comment]
Hi! 
Did you use the create hook? Read the HTTP request chapter in [the readme](https://github.com/arve0/feathers-custom-methods/blob/master/README.md)?

**Comment by leob** [comment]
Yes, I've read that ... aha, so you mean to say that you should just configure the hooks on "create" (rather than on the custom method), and then within the 'create' hook you branch on the value of the "method" parameter?

Yes that would work, and then I suppose it would also work if you use websockets rather than HTTP as the transport. Maybe it would be a good idea to add a note about this to the README?

**Comment by arve0** [comment]
Yes, that is correct, and yes, it should be in the readme. Leave this issue open until we fix the readme with a hook example.
